src/XMakeBuildEngine/Evaluation/IntrinsicFunctions.cs

```diff
@@ -364,6 +364,17 @@ internal static bool DoesTaskHostExist(string runtime, string architecture)
             return false;
         }
 
+        internal static bool IsOSPlatform(string os)
+        {
+            switch (os.ToLower())
+            {
+                case "windows": return NativeMethodsShared.IsWindows;
+                case "osx": return NativeMethodsShared.IsOSX;
+                case "unix": return NativeMethodsShared.IsUnixLike;
+                default: return false;
```

Comment:
Sure, but then we need to ensure that they support the same "set".

Looking at [this](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs) and [this](https://github.com/dotnet/corefx/blob/master/src/Native/configure.cmake), the name depends on `PAL_UNIX_NAME` set in the `configure.cmake` using `CMAKE_SYSTEM_NAME`.

[This](https://cmake.org/cmake/help/v3.4/variable/CMAKE_SYSTEM_NAME.html) and [this](https://cmake.org/cmake/help/v3.4/variable/CMAKE_HOST_SYSTEM_NAME.html#variable:CMAKE_HOST_SYSTEM_NAME) suggest that it uses `uname -s`, when available, for this. So, we could use `uname -s` too on non-windows systems. `Darwin`->`OSX` being the special case.

And maybe add a `Unix` too (== `IsUnixLike`) ?